libs/langchain-community/src/graphs/neo4j_graph.ts

```diff
@@ -211,36 +361,201 @@ export class Neo4jGraph {
+        outputDict[propName] = `{${returnClauses.pop()}}`;
+      }
+    } else {
+      matchClause += ` WITH n LIMIT 5`;
+
+      for (const prop of properties) {
+        const propName = prop.property;
+        const propType = prop.type;
+
+        const propIndex = this.structuredSchema?.metadata?.index.filter(
+          (el: Any) =>
+            el.label === labelOrType &&
+            el.properties[0] === propName &&
+            el.type === "RANGE"
+        );
+
+        if (propType === "STRING") {
+          if (
+            propIndex.length > 0 &&
+            propIndex[0].size > 0 &&
+            propIndex[0].distinctValues <= DISTINCT_VALUE_LIMIT
+          ) {
+            const distinctValuesPromise = await this.query(
+              `CALL apoc.schema.properties.distinct('${labelOrType}', '${propName}') YIELD value`
+            );
+            const distinctValues = distinctValuesPromise[0].value;
+            returnClauses.push(
+              `values: ${distinctValues}, distinct_count: ${distinctValues.length}`
+            );
+          } else {
+            withClauses.push(
+              `collect(distinct substring(n.\`${propName}\`, 0, 50)) AS \`${propName}_values\``
+            );
+            returnClauses.push(`values: ${propName}_values`);
+          }
+        } else if (
+          ["INTEGER", "FLOAT", "DATE", "DATE_TIME", "LOCAL_DATE_TIME"].includes(
+            propType
+          )
+        ) {
+          if (!propIndex) {
+            withClauses.push(
+              `collect(distinct toString(n.\`${propName}\`)) AS \`${propName}_values\``
+            );
+            returnClauses.push(`values: ${propName}_values`);
+          } else {
+            withClauses.push(`min(n.\`${propName}\`) AS \`${propName}_min\``);
+            withClauses.push(`max(n.\`${propName}\`) AS \`${propName}_max\``);
+            withClauses.push(
+              `count(distinct n.\`${propName}\`) AS \`${propName}_distinct\``
+            );
+            returnClauses.push(
+              `min: toString(\`${propName}_min\`), max: toString(\`${propName}_max\`), distinct_count: \`${propName}_distinct\``
+            );
+          }
+        } else if (propType === "LIST") {
+          withClauses.push(
+            `min(size(n.\`${propName}\`)) AS \`${propName}_size_min\`, max(size(n.\`${propName}\`)) AS \`${propName}_size_max\``
+          );
+          returnClauses.push(
+            `min_size: \`${propName}_size_min\`, max_size: \`${propName}_size_max\``
+          );
+        } else if (["BOOLEAN", "POINT", "DURATION"].includes(propType)) {
+          continue;
+        }
+
+        outputDict[propName] = `{${returnClauses.pop()}}`;
+      }
+    }
+
+    const withClause = `WITH ${withClauses.join(", ")}`;
+    const returnClause = `RETURN {${Object.entries(outputDict)
+      .map(([k, v]) => `\`${k}\`: ${v}`)
+      .join(", ")}} AS output`;
+
+    const cypherQuery = [matchClause, withClause, returnClause].join("\n");
```

Comment:
Triple checking that there's no injection risk here - has been a while since I've looked at Neo4j

Reply:
Yeah... actually user doesn't have any parameters they could inject... we just sample the database values with enhanced schema. But we try and escape most values anyway as a general guideline.